pkg/operator/osd/osd.go

```diff
@@ -107,8 +107,15 @@ func (c *Cluster) Start() error {
 
 			resources := k8sutil.MergeResourceRequirements(c.Storage.Nodes[i].Resources, c.resources)
 
+			var directories []Directory
+			if len(n.Directories) > 0 {
```

Comment:
this check looks similar to the work being done in the `resolveNode*` functions: https://github.com/rook/rook/blob/master/pkg/operator/osd/storage.go#L70

if you refactor so that the directories field is in the selection struct, then figuring out if the cluster wide or node specific directories should be used fits naturally in the `resolveNode*` functions.  then you wouldn't even have to make this determination here nor pass a directories arg to `makeReplicaSet`, it would have already figured it all out at resolve time a few lines above.

let me know if that's not clear.

Reply:
@jbw976 If I understand you correctly, the `directories` option I introduce here should be moved here: https://github.com/rook/rook/blob/master/pkg/operator/osd/spec.go#L54-L62 ?

Reply:
yes, i think that including it in the Selection struct makes sense as the directories list is just another way of selecting storage, and then the directories list doesn't have to be declared in multiple places.